patch-package
--- a/node_modules/@segment/analytics-react-native-firebase/RNAnalyticsIntegration-Firebase.podspec
+++ b/node_modules/@segment/analytics-react-native-firebase/RNAnalyticsIntegration-Firebase.podspec
@@ -25,4 +25,5 @@ Pod::Spec.new do |s|
   s.dependency          'Segment-Firebase'
   s.dependency          'Analytics'
   s.dependency          'React'
+  s.dependency          'RNAnalytics'
 end
--- a/node_modules/@segment/analytics-react-native-firebase/android/build.gradle
+++ b/node_modules/@segment/analytics-react-native-firebase/android/build.gradle
@@ -26,6 +26,10 @@ android {
         versionName '1.0'
 
     }
+
+    packagingOptions {
+        exclude 'META-INF/-no-jdk.kotlin_module'
+    }
 }
 
 repositories {
--- a/node_modules/@segment/analytics-react-native-firebase/ios/main.m
+++ b/node_modules/@segment/analytics-react-native-firebase/ios/main.m
@@ -8,7 +8,7 @@
 
 #import <React/RCTBridgeModule.h>
 #import <RNAnalytics/RNAnalytics.h>
-#import <Segment-Firebase/SEGFirebaseIntegrationFactory.h>
+#import <Segment_Firebase/SEGFirebaseIntegrationFactory.h>
 
 @interface RNAnalyticsIntegration_Firebase: NSObject<RCTBridgeModule>
 @end
