diff --git a/node_modules/@segment/analytics-react-native/ios/RNAnalytics/RNAnalytics.m b/node_modules/@segment/analytics-react-native/ios/RNAnalytics/RNAnalytics.m
index b3876d3..d2e0601 100644
--- a/node_modules/@segment/analytics-react-native/ios/RNAnalytics/RNAnalytics.m
+++ b/node_modules/@segment/analytics-react-native/ios/RNAnalytics/RNAnalytics.m
@@ -7,7 +7,8 @@
 
 #import "RNAnalytics.h"
 
-#import <Analytics/SEGAnalytics.h>
+// Remove patch once a release lands with https://github.com/segmentio/analytics-react-native/pull/234
+#import <Segment/SEGAnalytics.h>
 #import <React/RCTBridge.h>
 
 static NSMutableSet* RNAnalyticsIntegrations = nil;
