diff --git a/node_modules/bn.js/lib/bn.js b/node_modules/bn.js/lib/bn.js
index 855dfa6..5fa5fbd 100644
--- a/node_modules/bn.js/lib/bn.js
+++ b/node_modules/bn.js/lib/bn.js
@@ -48,10 +48,17 @@
   BN.BN = BN;
   BN.wordSize = 26;
 
-  var Buffer;
-  try {
-    Buffer = require('buffer').Buffer;
-  } catch (e) {
+  if (!Buffer) {
+    var Buffer;
+    if (global && global.Buffer) {
+      Buffer = global.Buffer
+    } else if (window && window.Buffer) {
+      Buffer = window.Buffer;
+    } else {
+      try {
+        Buffer = require("buffer").Buffer;
+      } catch (e) {}
+    }
   }
 
   BN.isBN = function isBN (num) {
