diff --git a/node_modules/react-native-reanimated/mock.js b/node_modules/react-native-reanimated/mock.js
index 2f2600d..d047c16 100644
--- a/node_modules/react-native-reanimated/mock.js
+++ b/node_modules/react-native-reanimated/mock.js
@@ -213,7 +213,11 @@ const Reanimated = {
   proc: (cb) => cb,
 
   useCode: NOOP,
+
   createAnimatedComponent: (Component) => Component,
+
+  interpolateColors: NOOP,
+
 };
 
 module.exports = {
