patch-package
new file mode 100644
--- /dev/null
+++ b/node_modules/react-native-secure-randombytes/RNRandomBytes.podspec
@@ -0,0 +1,18 @@
+require "json"
+package = JSON.parse(File.read(File.join(__dir__, "package.json")))
+
+Pod::Spec.new do |s|
+    s.name         = "RNRandomBytes"
+    s.version      = package['version']
+    s.summary      = package['description']
+
+    s.license      = package['license']
+    s.authors      = package['author']
+    s.homepage     = "https://github.com/WoeOm/react-native-secure-randombytes"
+    s.platform     = :ios, "9.0"
+
+    s.source       = { :git => "https://github.com/WoeOm/react-native-secure-randombytes" }
+    s.source_files  = "*.{h,m}"
+
+    s.dependency 'React'
+end
