diff --git a/node_modules/react-native-reanimated/android/build.gradle b/node_modules/react-native-reanimated/android/build.gradle
--- a/node_modules/react-native-reanimated/android/build.gradle
+++ b/node_modules/react-native-reanimated/android/build.gradle
@@ -56,8 +56,11 @@
 import org.apache.tools.ant.taskdefs.condition.Os
 import org.apache.tools.ant.filters.ReplaceTokens

+def reactNative = new File("$projectDir/../../../node_modules/react-native")
+def reactNativeThirdParty = new File("$reactNative/ReactAndroid/src/main/jni/third-party")
+
 def reactProperties = new Properties()
-file("$projectDir/../node_modules/react-native/ReactAndroid/gradle.properties").withInputStream { reactProperties.load(it) }
+file("$reactNative/ReactAndroid/gradle.properties").withInputStream { reactProperties.load(it) }

 def BOOST_VERSION = reactProperties.getProperty("BOOST_VERSION")
 def DOUBLE_CONVERSION_VERSION = reactProperties.getProperty("DOUBLE_CONVERSION_VERSION")
@@ -71,9 +74,6 @@
 def downloadsDir = new File("$buildDir/downloads")
 def thirdPartyNdkDir = new File("$buildDir/third-party-ndk")

-def reactNative = new File("$projectDir/../node_modules/react-native")
-def reactNativeThirdParty = new File("$reactNative/ReactAndroid/src/main/jni/third-party")
-
 // You need to have following folders in this directory:
 //   - boost_1_63_0
 //   - double-conversion-1.1.6
@@ -387,13 +387,13 @@
             "NDK_OUT=" + temporaryDir,
             "NDK_LIBS_OUT=$buildDir/reanimated-ndk/all",
             "THIRD_PARTY_NDK_DIR=$buildDir/third-party-ndk",
-            "JSI_PATH=$projectDir/../node_modules/react-native/ReactCommon/jsi",
+            "JSI_PATH=$reactNative/ReactCommon/jsi",
             "REACT_COMMON_DIR=$reactNative/ReactCommon",
             "REACT_SRC_DIR=$reactNative/ReactAndroid/src/main/java/com/facebook/react",
             "FIRST_PARTY=$reactNativeThirdParty/../first-party",
             "REACT_NATIVE_JNI=$reactNative/ReactAndroid/src/main/jni/react/jni",
             "REACT_NATIVE_REACT=$reactNative/ReactAndroid/src/main/jni/react",
-            "HERMES_ENGINE=$projectDir/../node_modules/hermes-engine",
+            "HERMES_ENGINE=$projectDir/../../../node_modules/hermes-engine",
             "VM=HERMES",
             "-C", file("src/main/JNI").absolutePath,
             "--jobs", project.findProperty("jobs") ?: Runtime.runtime.availableProcessors()
@@ -448,3 +448,3 @@
     extractHeaders("com.facebook.fbjni:fbjni:0.0.2:headers")
     extractJNI("com.facebook.fbjni:fbjni:0.0.2")
 }
