patch-package
--- a/node_modules/react-native-secure-randombytes/RNRandomBytes.m
+++ b/node_modules/react-native-secure-randombytes/RNRandomBytes.m
@@ -20,6 +20,11 @@ RCT_EXPORT_MODULE()
 
 @synthesize bridge = _bridge;
 
++ (BOOL)requiresMainQueueSetup
+{
+    return YES;
+}
+
 RCT_EXPORT_METHOD(randomBytes:(NSUInteger)length
                   callback:(RCTResponseSenderBlock)callback)
 {
new file mode 100644
--- /dev/null
+++ b/node_modules/react-native-secure-randombytes/RNRandomBytes.podspec
@@ -0,0 +1,18 @@
+require "json"
+package = JSON.parse(File.read(File.join(__dir__, "package.json")))
+
+Pod::Spec.new do |s|
+    s.name         = "RNRandomBytes"
+    s.version      = package['version']
+    s.summary      = package['description']
+
+    s.license      = package['license']
+    s.authors      = package['author']
+    s.homepage     = "https://github.com/WoeOm/react-native-secure-randombytes"
+    s.platform     = :ios, "9.0"
+
+    s.source       = { :git => "https://github.com/WoeOm/react-native-secure-randombytes" }
+    s.source_files  = "*.{h,m}"
+
+    s.dependency 'React'
+end
