patch-package
--- a/node_modules/tslint/lib/linter.js
+++ b/node_modules/tslint/lib/linter.js
@@ -193,7 +193,7 @@ var Linter = /** @class */ (function () {
                 var oldSource = fs.readFileSync(filePath, "utf-8");
                 fileNewSource = rule_1.Replacement.applyFixes(oldSource, fileFixes);
             }
-            fs.writeFileSync(filePath, fileNewSource);
+            fs.writeFileSync(path.resolve(filePath), fileNewSource);
             _this.updateProgram(filePath);
         });
         return source;
