diff --git a/node_modules/websocket/lib/WebSocketFrame.js b/node_modules/websocket/lib/WebSocketFrame.js
index 859e879..92d0bbb 100644
--- a/node_modules/websocket/lib/WebSocketFrame.js
+++ b/node_modules/websocket/lib/WebSocketFrame.js
@@ -253,7 +253,7 @@ WebSocketFrame.prototype.toBuffer = function(nullMask) {
     }
 
     if (this.mask) {
-        maskKey = nullMask ? 0 : (Math.random()*0xFFFFFFFF) | 0;
+        maskKey = nullMask ? 0 : (Math.random()*0xFFFFFFFF) >>> 0;
         this.maskBytes.writeUInt32BE(maskKey, 0, true);
         
         // write the mask key
